docs/learning/frontend/Extensions.md

```diff
@@ -109,55 +132,238 @@ The only requirement when creating an Extension is to define a default function.
 The default function will execute on the Extension's activation event.
 The function would run immediately if onStartup was specified as the Activation Event.
 For example:
+
 ```tsx
 // src/index.ts
 export default function main() {
   console.log("Hello from Extension!");
 }
 ```
-The above would print "Hello from Extension!". We can extend functionality by adding, for example, a tool to print to the console whenever it is clicked:
+
+The above would print "Hello from Extension!".
+
+We can extend functionality by adding, for example, adding some decorators representing IoT devices, and when clicked will navigate to a specific view.
+First, we will query for all ceilings, walls, windows, etc, and hide them from the view.
+Then, we will query for all IoT devices, and add a decorator for each one.
+
+```ts
```

Comment:
Having full-blown multi-file sample code in a markdown file is not ideal...could it be moved to, e.g., example-code so we can confirm it compiles and continues to do so in future?